community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java

```diff
@@ -544,28 +543,11 @@
     public static final Setting<Integer> batch_inserter_batch_size = setting( "unsupported.tools.batch_inserter.batch_size", INTEGER,
             "10000" );
 
-    public enum LabelIndex
-    {
-        /**
-         * Native label index. Generally the best option.
-         */
-        NATIVE,
-
-        /**
-         * Label index backed by Lucene.
-         */
-        LUCENE,
-
-        /**
-         * Selects which ever label index is present in a store, or the default (NATIVE) if no label index present.
-         */
-        AUTO
-    }
-
+    // todo Can we simply remove this setting or do we need to depricate it?
```

Comment:
You have to revert everything in this file and add `@Deprecated` to it since its public, yes, even the enum.
What you could do is change `options( enumNames( LabelIndex.class ), true )` to `optionsIgnoreCase( "native" )` which would make any other setting than `native` fail with a nice error.